### DISCUSSION TITLE
Custom environment/schema path for CLI commands?

### DISCUSSION THREAD

**Discussion by quintal-william** [prisma/prisma#3013] [Q&A] [answered]
Hello Prisma team! 

I have a question regarding the CLI (mainly the `generate` and `migrate` commands).
Is it/will it be possible to supply a custom path to the schema and/or environment file? I am working on a full-stack Nextjs project where I'd like a .env file to be at the root and the schema to be in a src folder. This root .env file will not only contain prisma variables, like `DATABASE_URL`, but also variables for the rest of the application, like `FIREBASE_KEY`.

**Current situation:**
When running `prisma generate` or `prisma migrate save --experimental`, it seems to look for the files it needs in the `/prisma` directory and only there.

**Desired situation:**
The `generate` and `migrate` commands supply flags like `--env` and `--schema`, which can be used to specify the location of the files, such as: `prisma migrate save --experimental --env=/.env` or `prisma generate --env=/environments/.env.prisma --schema=/src/prisma/schema.prisma`.

I saw that the team was already working on the functionality of importing prisma schemas into other prisma schemas, which is a feature I'm looking forward to very much! If there is any other way this effect can be achieved, then I'd love to know it :)

**Comment by ryands17** [comment, marked as answer]
Hey @waffl :wave: 

For now, you can use [this](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema#using-environment-variables) workaround as described in the docs to use the `.env` of your choice.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
